jobs/mongodb_backup_job.go

```diff
@@ -147,141 +167,92 @@ func createDBURL(dbConfig DBConnConfig) *url.URL {
 	}
 }
 
-func (j *MongoDBBackupJob) startBackup(ctx context.Context) error {
+func (j *MongoDBBackupJob) startBackup(ctx context.Context) (*pbmBackup, error) {
 	j.l.Info("Starting backup.")
+	var result pbmBackup
 
-	nCtx, cancel := context.WithTimeout(ctx, cmdTimeout)
-	defer cancel()
-
-	output, err := exec.CommandContext(nCtx, pbmBin, "backup", "--mongodb-uri="+j.dbURL.String()).CombinedOutput() // #nosec G204
-
-	if err != nil {
-		return errors.Wrapf(err, "pbm backup error: %s", string(output))
+	if err := getPBMOutput(ctx, j.dbURL, &result, "backup"); err != nil {
+		return nil, err
 	}
 
-	return nil
+	return &result, nil
 }
 
-func checkRunningPBMOperations(ctx context.Context, l logrus.FieldLogger, dbURL *url.URL) (bool, error) {
-	l.Debug("Checking running pbm operations.")
-
-	nCtx, cancel := context.WithTimeout(ctx, cmdTimeout)
-	defer cancel()
-
-	output, err := exec.CommandContext(nCtx, pbmBin, "status", "--mongodb-uri="+dbURL.String()).CombinedOutput() // #nosec G204
-	if err != nil {
-		return false, errors.Wrapf(err, "pbm status error: %s", string(output))
+func (j *MongoDBBackupJob) streamLogs(ctx context.Context, send Send, name string, backupFinished <-chan struct{}) error {
+	var (
+		err        error
+		backupDone bool
+		logs       []pbmLogEntry
+		buffer     bytes.Buffer
+		skip       int
+		lastLog    pbmLogEntry
+	)
+	j.logChunkID = 0
+	finished := func() bool {
+		if lastLog.Msg == "backup finished" {
```

Comment:
Probably not needed anymore.
Logs are not delayed, and we keep still little delay just in case.
So I suggest, just to cancel context after this little delay and remove this.

Reply:
Agree